documentation/abnormal_termination/step2/using_diagnostic_report.md

```diff
@@ -8,15 +8,70 @@ configuration and component version issues with in your process.
 
 ## How To
 
-//TODO, raw notes from deep dive:
-
 1. Open the generated report (either from file or from console)
-2. Take a look to allocation section in the report
-3. Look for the old_space
-    - used is high
-    - available is very low
-4. Look javascriptStack to see the victim of the memory leak which can be the allocation pattern and source to find what causes it (JavaScript stack trace won’t always point to the right frame, can be confusing)
-5. -> see memory related user journeys to identify exact cause
+
+  Report can be generated using the following cli flags: `node --report-on-fatal-error <app.js>`.
+  Report can be generated using the following programatic configuration:
+  ```js
+  process.report...
+  ```
+
+2. Open the generated report which is in the format `report.20200410.114639.4623.0.001.json`. 
+
+3. Take a look at the `javascriptHeap` section in the report
+  The `javascriptHeap` will give the details of memory allocation. It is split into 6 regions:
+    * read_only_space
+    * new_space
+    * old_space
+    * code_space
+    * map_space
+    * large_object_space
+
+4. Inpsect each spaces. For each space,
+  ```js
+  "old_space": {
+      "memorySize": 5332910528,
+      "committedMemory": 5332846624,
+      "capacity": 5332901608,
+      "used": 5332770856,
+      "available": 130752
+  },
+  ```
+ * used is high : Out of allocated  memory 5332910528 bytes, 5332770856 bytes are used.
+ * available is very low : 130752 bytes.
+
+5. Look at `javascriptStack` to see the victim of the memory leak which can be the allocation pattern and /or the actual source. Note that `javaScriptStack` will not always point to the right frame.
```

Comment:
Maybe `javascriptStack` can be inspected first to get an overview of the cause of the termination?

Reply:
@legendecas, re arranged it. PTAL.